### DISCUSSION TITLE
Delay time (≈28 sec) before an email is processed and sent out

### DISCUSSION THREAD

**Discussion by michael-v-bloom** [modoboa/modoboa#2420] [General]
Hello everyone,

I have searched everywhere for an answer, but find myself finally asking for your help and input here.

_**We have a working Modoboa installation**_ (v. 1.17.0) on Ubuntu 20.04 (LTS) x64 (hosted on DigitalOcean). The current installation doesn't show any errors, and the emails are sent out and received correctly.

**The main issue:** 
Each time we send out an email to another email account, the email message is registered by POSTFIX. Then nothing happens for about 28-30 seconds, and only after this (delay or queue?) time POSTFIX logs the connection to the specified email server and sends out the mentioned email message.

Here's the log of the issue (note difference of time in bold):

`Dec  5 17:34:15 mail postfix/submission/smtpd[316066]: connect from hidden-ip-address[XX.XXX.XXX.XX]`

` Dec  5 17:34:15 mail postfix/submission/smtpd[316066]: Anonymous TLS connection established from hidden-ip-address[XX.XXX.XXX.XX]: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange ECDHE (P-256) server-signature RSA-PSS (2048 bits) server-digest SHA256` 

` Dec  5 17:34:15mail postfix/submission/smtpd[316066]: NOQUEUE: client=hidden-ip-address[XX.XXX.XXX.XX], sasl_method=PLAIN, sasl_username=senderemail@dezeamail.com` 

` Dec  5 17:34:48 mail postfix/smtpd[316091]: connect from localhost[127.0.0.1]` 

` Dec  5 17:34:48 mail postfix/smtpd[316091]: 5410A13C162: client=localhost[127.0.0.1], orig_client=hidden-ip-address[XX.XXX.XXX.XX]` 

` Dec  5 17:34:48 mail postfix/cleanup[316092]: 5410A13C162: message-id=<cd320a5e-6227-406e-f349-ed275609718d@dezeamail.com>` 

` Dec  5 17:34:48 mail postfix/smtpd[316091]: disconnect from localhost[127.0.0.1] ehlo=1 xforward=1 mail=1 rcpt=1 data=1 quit=1 commands=6` 

` Dec  5 17:34:48 mail postfix/qmgr[190225]: 5410A13C162: from=<senderemail@dezeamail.com>, size=1360, nrcpt=1 (queue active)` 

` Dec  5 17:34:48 mail postfix/submission/smtpd[316066]: proxy-accept: END-OF-MESSAGE: 250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 5410A13C162; from=<senderemail@dezeamail.com> to=<receiveremail@gmail.com> proto=ESMTP helo=<[192.168.0.2]> sasl_username=<senderemail@dezeamail.com>` 

` Dec  5 17:34:48 mail postfix/smtp[316093]: connect to gmail-smtp-in.l.google.com[2a00:1450:4013:c16::1a]:25: Network is unreachable` 

` Dec  5 17:34:48 mail postfix/submission/smtpd[316066]: disconnect from hidden-ip-address[XX.XXX.XXX.XX] ehlo=2 starttls=1 auth=1 mail=1 rcpt=1 data=1 quit=1 commands=8` 

` Dec  5 17:34:48 mail postfix/smtp[316093]: Trusted TLS connection established to gmail-smtp-in.l.google.com[142.251.31.27]:25: TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits) key-exchange X25519 server-signature ECDSA (P-256) server-digest SHA256` 

` Dec  5 17:34:49 mail postfix/smtp[316093]: 5410A13C162: to=<receiveremail@gmail.com>, relay=gmail-smtp-in.l.google.com[142.251.31.27]:25, delay=0.89, delays=0.08/0.01/0.26/0.53, dsn=2.0.0, status=sent (250 2.0.0 OK  1638725689 h12si22621547edb.432 - gsmtp)` 

` Dec  5 17:34:49 mail postfix/qmgr[190225]: 5410A13C162: removed` 

I have been looking for a solution to somehow skip this delay in the POSTFIX configuration files **master.cf** and **main.cf** by studying the documentation... But, no solution found. 

I have also found in other posts here (on Modoboa Github community) about the possible Amavis message scanning time that could create this type of delays... But, also no solution identified.

Hopefully, you have a hint where to search. Thanks!

Best regards,
Michael Bloom

**Comment by michael-v-bloom** [comment]
In case someone experiences the same issue! 

**THE SOLUTION:**
- Upgraded from 2GB to 4GB of RAM on my VM

**THE RESULT:**
1. This has accelerated the sent out of emails by 10x
2. This has also fixed the `***UNCHECKED***` message from emails' Subject.

**Reply by MrGeneration** [reply]
Running modoboa with all full fledged services may even force you to aim for 8GB ram.
My rather IDLE setup currently uses 4,7 GB of ram - every mail that comes in raises the required memory temporary. I've seen bumps to up to 7GB. Depends on your load through.

**Reply by michael-v-bloom** [reply]
Thanks for the input. Will take it in consideration.
